src/RageSoundPosMap.cpp

```diff
@@ -133,38 +118,34 @@ std::int64_t pos_map_queue::Search( std::int64_t iSourceFrame, bool *bApproximat
 		return 0;
 	}
 
-	/* iSourceFrame is probably in pos_map.  Search to figure out what position
-	 * it maps to. */
-	std::int64_t iClosestPosition = 0, iClosestPositionDist = INT_MAX;
-	const pos_map_t *pClosestBlock = &*m_pImpl->m_Queue.begin(); /* print only */
+	// iSourceFrame is probably in pos_map.  Search to figure out what position it maps to.
+	std::int64_t iClosestPosition = 0, iClosestPositionDist = std::numeric_limits<int64_t>::max();
 	for (pos_map_t const &pm : m_pImpl->m_Queue)
 	{
+		// Loop over the queue until we know generally where iSourceFrame is
 		if( iSourceFrame >= pm.m_iSourceFrame &&
 			iSourceFrame < pm.m_iSourceFrame+pm.m_iFrames )
 		{
-			/* iSourceFrame lies in this block; it's an exact match.  Figure
-			 * out the exact position. */
-			int iDiff = int(iSourceFrame - pm.m_iSourceFrame);
-			iDiff = std::lrint( iDiff * pm.m_fSourceToDestRatio );
+			// If we are in the correct block, calculate its current position
+			std::int64_t iDiff = static_cast<std::int64_t>(iSourceFrame - pm.m_iSourceFrame);
+			iDiff = static_cast<int64_t>(( iDiff * pm.m_fSourceToDestRatio) + 0.49 ); 
 			return pm.m_iDestFrame + iDiff;
```

Comment:
Why are these 0.49 now? 0.5 is correct to replicate a round.

e.g. 0.5 + 0.49 = 0.99, which would cast down to 0, but 0.5 should round up to 1

Reply:
> Why are these 0.49 now? 0.5 is correct to replicate a round.
> 
> e.g. 0.5 + 0.49 = 0.99, which would cast down to 0, but 0.5 should round up to 1

Thank you for the review!

This was done to replicate the behavior of `lrint()` exactly. This sample code explains the reason I made this change:  https://ideone.com/Ngu8BO

```
#include <iostream>
#include <cmath>

using namespace std;

float tm = 6.999;
float tmm = 6.500;
float tmr = 6.199;

int main() {
	
	int64_t seconds = static_cast<int64_t>(tm + 0.49);
	std::cout << "Static cast (+.49) of tm (" << tm << ") is " << seconds << std::endl;
	
	int64_t secondsq = static_cast<int64_t>(tm + 0.5);
	std::cout << "Static cast (+.5) of tm (" << tm << ") is " << secondsq << std::endl;
	
	int64_t secondX = std::lrint(tm);
	std::cout << "std::lrint of tm (" << tm << ") is " << secondX << std::endl;
	
	int64_t secondsm = static_cast<int64_t>(tmm + 0.49);
	std::cout << "Static cast (+.49) of tmm (" << tmm << ") is " << secondsm << std::endl;
	
	int64_t secondsmq = static_cast<int64_t>(tmm + 0.5);
	std::cout << "Static cast (+.5) of tmm (" << tmm << ") is " << secondsmq << std::endl;
	
	int64_t secondXm = std::lrint(tmm);
	std::cout << "std::lrint of tmr (" << tmm << ") is " << secondXm << std::endl;
	
	int64_t secondsr = static_cast<int64_t>(tmr + 0.49);
	std::cout << "Static cast (+.49) of tmr (" << tmr << ") is " << secondsr << std::endl;
	
	int64_t secondsrq = static_cast<int64_t>(tmr + 0.5);
	std::cout << "Static cast (+.5) of tmr (" << tmr << ") is " << secondsrq << std::endl;
	
	int64_t secondXr = std::lrint(tmr);
	std::cout << "std::lrint of tmr (" << tmr << ") is " << secondXr << std::endl;
	
	return 0;
}
```

```
Static cast (+.49) of tm (6.999) is 7
Static cast (+.5) of tm (6.999) is 7
std::lrint of tm (6.999) is 7

Static cast (+.49) of tmm (6.5) is 6
Static cast (+.5) of tmm (6.5) is 7
std::lrint of tmr (6.5) is 6

Static cast (+.49) of tmr (6.199) is 6
Static cast (+.5) of tmr (6.199) is 6
std::lrint of tmr (6.199) is 6
```